packages/api/src/record/service.ts

```diff
@@ -25,3 +31,40 @@ export const getRecordById = async (requestQuery: {
   }));
   return records;
 };
+
+function buildPatchQuery(patchRecordRequest: RecordColumnsForUpdate): string {
+  const updates = Object.entries(patchRecordRequest)
+    .filter(([key, value]) => value !== undefined && key !== "recordId")
+    .map(([key, _]) => `${key} = :${key}`);
+
+  if (updates.length === 0) {
+    throw new createError.BadRequest("Request cannot be empty");
+  }
+
+  const query = `
+    UPDATE record
+    SET ${updates.join(", ")}
+    WHERE recordid = :recordId
+    RETURNING record.recordid AS "recordId"
+  `;
+
+  return query.trim();
+}
+
+export const patchRecord = async (
+  recordId: string,
+  recordData: PatchRecordRequest
+): Promise<string> => {
+  const dto = requestFieldsToDatabaseFields(recordData, recordId);
```

Comment:
This should renamed now that the object assigned to this variable is not called a DTO.